### DISCUSSION TITLE
Web browser paths to global variable?

### DISCUSSION THREAD

**Discussion by nobody43** [roddhjav/apparmor.d#127] [Q&A] [answered]
There are many programs that needs the ability to launch web browser for them. Now we have only specific Firefox profile to point to. But users might have different installation of FF or they might use different browser altogether.
How this could be fixed?

**Comment by roddhjav** [comment]
This is usually handled by `child-open`. See https://apparmor.pujol.io/development/structure/#children-profiles 
However, this only cover the program started by tool like xdg-open.

**Comment by nobody43** [comment]
OK, thanks. Any suggestions on how to refactor profiles like [thunderbird](https://github.com/roddhjav/apparmor.d/blob/main/apparmor.d/groups/apps/thunderbird) and [qbittorrent](https://github.com/roddhjav/apparmor.d/blob/main/apparmor.d/profiles-m-r/qbittorrent)?

**Comment by roddhjav** [comment, marked as answer]
Thunderbird use an program like `xdg-open`, simply replace `rCx -> open,` by `rPx -> child-open` (and remove the `open` subprofile) and that should be OK. For qbittorrent, they seem to open the program directly so there is no solution right now. The way I see a solution would be to create an `open` abstraction that allow to start the same program than in `child-open`.
